src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -1156,14 +769,11 @@ private String getConcurrentModificationMessage() {
     public Collection<E> lock(Iterable<E> entities) {
         Map<Serializable, E> entityMap = new TreeMap<>();
 
-        SessionImpl session = (SessionImpl) this.currentSession();
-        ClassMetadata metadata = session.getSessionFactory().getClassMetadata(this.entityType);
-
         // Step through and toss all the entities into our TreeMap, which orders its entries using
         // the natural order of the key. This will ensure that we have our entity collection sorted
         // by entity ID, which should help avoid deadlock by having a deterministic locking order.
         for (E entity : entities) {
-            entityMap.put(metadata.getIdentifier(entity, session), entity);
```

Comment:
This seems to also have been written this way because we use the 'uuid' field as the primary key for Content and Product entities. However, if E ends up being either Content/Product, now after the Entity Layering change, these will end up being unique per the 'id' field, so we should still be getting a consistent locking order here. 

@Ceiu  is my assumption correct here?^

Reply:
Nope. With layering+namespaces, the constraint with IDs is on namespace+ID, so this runs into the problem of potentially pulling more than one thing.

Reply:
I can't seem to find a way to get this metadata through EntityManager, so this may be a case where we'll have to use the Hibernate session still

Reply:
In pure JPA, it's behind the `PersistenceUnitUtil`, which itself is fetchable through the `EntityManagerFactory`.

Reply:
done